apps/src/templates/sectionsRefresh/CoteacherSettings.jsx

```diff
@@ -33,36 +74,152 @@ export default function CoteacherSettings({
       return;
     }
     if (
-      (sectionInstructors &&
-        sectionInstructors.some(
-          instructor => instructor.instructorEmail === newEmail
-        )) ||
-      coteachersToAdd.some(email => email === newEmail)
+      coteachers.some(instructor => instructor.instructorEmail === newEmail)
     ) {
       setAddError(i18n.coteacherAddAlreadyExists({email: newEmail}));
       return;
     }
 
-    addCoteacher(newEmail);
+    setCoteachersToAdd(existing => [newEmail, ...existing]);
     setAddError('');
     setInputValue('');
   };
 
+  const handleInputChange = event => {
+    setInputValue(event.target.value);
+  };
+
   const handleSubmitAddEmail = e => {
     if (e.key === 'Enter') {
       e.preventDefault();
       handleAddEmail(e);
     }
   };
 
-  // coteacher count is teachers to add + the existing teachers - the primary teacher.
-  // If sectionInstructors is populated, we assume one of them is the primary teacher.
-  const existingCoteachers = sectionInstructors
-    ? sectionInstructors.length - 1
-    : 0;
-  const numCoteachers = existingCoteachers + coteachersToAdd.length;
+  const pill = (text, className, icon) => {
+    return (
+      <div className={classNames(className, styles.tablePill)}>
+        <StrongText>
+          <FontAwesome icon={icon} className={styles.tablePillIcon} />
+          {text}
+        </StrongText>
+      </div>
+    );
+  };
```

Comment:
I tend to prefer this style for one-statement functions: two fewer lines and one fewer level of indentation.

```suggestion
  const pill = (text, className, icon) => (
    <div className={classNames(className, styles.tablePill)}>
      <StrongText>
        <FontAwesome icon={icon} className={styles.tablePillIcon} />
        {text}
      </StrongText>
    </div>
  );
```